lib/kvdb/ikvdb.c

```diff
@@ -372,12 +376,38 @@ ikvdb_create(const char *kvdb_home, struct kvdb_cparams *params)
     return err;
 }
 
+static merr_t
+ikvdb_is_pmem_only(const char *kvdb_home, const struct kvdb_meta *meta, bool *pmem_only)
```

Comment:
Instead of this function existing, can we not stash whether it is pmem only on the struct? This seems rather expensive

Reply:
I see this is for un-opened databases however.

Reply:
Yes, this is stashed in ikvdb_impl struct. The dax filesystem check is done once at kvdb create, kvdb open and storage add. We can add a new type field to the kvdb.meta file to record whether a KVDB is standard/pmem-only and avoid this check at kvdb open, but I won't go there until it's proven to be expensive. So far I didn't find any noticeable lag caused by this check.